src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -1692,16 +1822,73 @@ def read_global_sensitivity_analysis_meta_data(
     )
 
 
-def read_global_sensitivity_analysis_time_set(
+def read_uncertainty_propagation_analyses_meta_data(
     result_directory: Path,
-) -> Optional[numpy.array]:
+) -> Optional[UncertaintyPropagationAnalysesMetaData]:
+    """
+    Read the uncertainty propagation analyses metadata persisted in a result file.
     """
-    Get the time set for sensitivity analysis results.
+
+    return UncertaintyPropagationAnalysesMetaData.get_metadata_from_dir(
+        result_directory=result_directory
+    )
+
+
+def read_uncertainty_propagation_results(
+    metadata: UncertaintyPropagationAnalysesMetaData, results_key: str
+) -> Optional[UPResult]:
+    """
+    Get the uncertainty propagation results.
+
+    :param metadata:
+        The uncertainty propagation metadata previously read.
+
+    :param results_key:
+        The result key as follows: <property_id@trend_id>
+    """
+    meta = metadata.items.get(results_key)
+    if not meta:
+        return None
+
+    with open_result_file(metadata.result_directory) as file:
+        up_group = file[UNCERTAINTY_PROPAGATION_GROUP_NAME]
+        realization_output_samples = up_group[
+            UNCERTAINTY_PROPAGATION_DSET_REALIZATION_OUTPUS
+        ]
+
+        realization_outputs = [
+            realization_output_samples[sample_index][qoi_index]
+            for qoi_index, sample_index in meta.sample_indexes
+        ]
+        mean_result = up_group[UNCERTAINTY_PROPAGATION_DSET_MEAN_RESULT][
+            meta.result_index
+        ]
+        std_result = up_group[UNCERTAINTY_PROPAGATION_DSET_STD_RESULT][
+            meta.result_index
+        ]
+        return UPResult(
+            realization_output=realization_outputs,
+            mean_result=mean_result,
+            std_result=std_result,
+            category=meta.category,
+            unit=meta.unit,
+        )
+
+
+def read_uq_time_set(result_directory: Path, group_name: str) -> Optional[numpy.array]:
+    """
+    Get the time set for based uq analysis results.
```

Comment:
This doesn't apply to all UQ analysis, so my recommendation is to make it clear in the docstrings. Same for the `BaseUQMetadata` class.

```suggestion
def read_uq_time_set(result_directory: Path, group_name: str) -> Optional[numpy.array]:
    """
    Get the time set for uq-based analysis results (Global Sensitivity Analysis or Uncertainty Propagation).
```